src/vcpkg/tools.cpp

```diff
@@ -62,82 +64,115 @@ namespace vcpkg
+
+        Json::Reader r(origin);
+        auto maybe_tool_data = r.visit(as_value, ToolDataArrayDeserializer::instance);
+        if (!r.errors().empty() || !r.warnings().empty())
+        {
+            return r.join();
+        }
+
+        if (auto tool_data = maybe_tool_data.get())
+        {
+            return *tool_data;
+        }
+
+        return msg::format_error(msgErrorWhileParsingToolData, msg::path = origin);
+    }
+
+    static ExpectedL<std::vector<ArchToolData>> parse_tool_data_file(const Filesystem& fs, Path path)
+    {
+        std::error_code ec;
+        auto contents = fs.read_contents(path, ec);
+        if (ec)
+        {
+            return format_filesystem_call_error(ec, __func__, {path});
+        }
+
+        return parse_tool_data(contents, path);
+    }
+
+    const ArchToolData* get_raw_tool_data(const std::vector<ArchToolData>& tool_data_table,
+                                          StringView toolname,
+                                          const CPUArchitecture arch,
+                                          StringView os)
+    {
+        const ArchToolData* default_tool = nullptr;
+        for (auto&& tool_candidate : tool_data_table)
+        {
+            if (tool_candidate.tool == toolname && tool_candidate.os == os)
+            {
+                if (!tool_candidate.arch)
+                {
+                    if (!default_tool)
+                    {
+                        default_tool = &tool_candidate;
+                    }
+                }
+                else if (arch == *tool_candidate.arch.get())
+                {
+                    return &tool_candidate;
+                }
+            }
+        }
+        return default_tool;
+    }
+
+    static Optional<ToolData> get_tool_data(const std::vector<ArchToolData>& tool_data_table, StringView tool)
+    {
+        auto hp = get_host_processor();
 #if defined(_WIN32)
-        return parse_tool_data_from_xml(XML, XML_PATH, tool, "windows");
+        auto data = get_raw_tool_data(tool_data_table, tool, hp, "windows");
 #elif defined(__APPLE__)
-        return parse_tool_data_from_xml(XML, XML_PATH, tool, "osx");
+        auto data = get_raw_tool_data(tool_data_table, tool, hp, "osx");
 #elif defined(__linux__)
-        return parse_tool_data_from_xml(XML, XML_PATH, tool, "linux");
+        auto data = get_raw_tool_data(tool_data_table, tool, hp, "linux");
 #elif defined(__FreeBSD__)
-        return parse_tool_data_from_xml(XML, XML_PATH, tool, "freebsd");
+        auto data = get_raw_tool_data(tool_data_table, tool, hp, "freebsd");
 #elif defined(__OpenBSD__)
-        return parse_tool_data_from_xml(XML, XML_PATH, tool, "openbsd");
+        auto data = get_raw_tool_data(tool_data_table, tool, hp, "openbsd");
 #else
         return nullopt;
 #endif
-    }
-
-    Optional<ToolData> parse_tool_data_from_xml(StringView XML, StringView XML_PATH, StringView tool, StringView os)
-    {
-        static const char* XML_VERSION = "2";
```

Comment:
New file format is missing this document version field. Is this a feature we want to preserve?